docs/Cluster.md

```diff
@@ -0,0 +1,177 @@
+- use a load balancer to hide all your `frontend` servers under the same ip address
+- set firewall rules to disable lightning connections (port 9735) on your `backend` server, so all connections go through the `frontend`
+- enable [monitoring](Monitoring.md)
+- on AWS, use AWS Secrets Manager (see [AWS deployment](#aws-deployment))
+
+#### Enable encrypted communication for the cluster
+
+We use a self-signed certificate, which offers a good compromise. More advanced options are available, see [akka doc](https://doc.akka.io/docs/akka/current/remoting-artery.html#remote-security).
+> Have a single set of keys and a single certificate for all nodes and disable hostname checking
+> - The single set of keys and the single certificate is distributed to all nodes. The certificate can be self-signed as it is distributed both as a certificate for authentication but also as the trusted certificate.
+> - If the keys/certificate are lost, someone else can connect to your cluster.
+> - Adding nodes to the cluster is simple as the key material can be deployed / distributed to the new node.
+
+Generate a self-signed certificate (set a strong password):
+```shell
+$ keytool -genkeypair -v \
+          -keystore akka-cluster-tls.jks \
+          -dname "O=ACME, C=FR" \
+          -keypass:env <password> \
+          -storepass:env <password> \
+          -keyalg RSA \
+          -keysize 4096 \
+          -validity 9999
+```
+
+Copy the resulting certificate to your `.eclair` directory:
+```shell
+$ cp akka-cluster-tls.jks ~/.eclair
+```
+
+Add this to your `eclair.conf`:
+```
+AKKA_TLS_PASSWORD=<password>
+akka.remote.artery.transport = "tls-tcp"
+```
+
+### AWS Deployment
+
+For convenience, we provide a prebuilt AWS Beanstalk bundle for the `frontend` (choose a WebServer environment type, and Java platform).
+
+You can run it as-is for testing.
+
+#### TLS encryption
+
+If you intend to use it in production, you need to enable encryption with your own certificate:
+1. Follow the procedure above to generate your `akka-tls.jks`
+
+2. We recommend forking the project and build your own bundle:
+```shell
+$ git clone git@github.com:ACINQ/eclair.git
+$ vi eclair-core/src/main/reference.conf # set akka.remote.artery.transport = "tls-tcp"
+$ cp akka-cluster-tls.jks eclair-front/modules/awseb/ # copy the file you generated
+$ vi eclair-front/modules/awseb.xml # uncomment the relevant parts
+$ mvn package -DskipTests
+```
+Alternatively, you can also edit the existing bundle and manually add the `akka-cluster-tls.jks` file to the root of the zip archive. You will also need to set `akka.remote.artery.transport=tls-tcp` at runtime. 
+
+#### Private key
+
+In production, we highly recommend using AWS Secrets manager to provide the node private key. This is done by setting `eclair.front.priv-key-provider=aws-sm`. Default secret name is "node-priv-key", but it is configurable with `eclair.front.aws-sm.aws-sm.priv-key-name`
```

Comment:
Is the duplicate `aws-sm.aws-sm` expected?

Reply:
Good catch!

```suggestion
In production, we highly recommend using AWS Secrets manager to provide the node private key. This is done by setting `eclair.front.priv-key-provider=aws-sm`. Default secret name is "node-priv-key", but it is configurable with `eclair.front.aws-sm.priv-key-name`
```